### ISSUE TITLE
Timestamps in conversations on Android app show "now" instead of accurate time

### ISSUE THREAD

**Issue by perrymike15** [klinker-apps/messenger-issues#721]
- [ ] I have [searched](https://github.com/klinker-apps/messenger-issues/issues), by keyword, for my issue/request to verify that there aren't any open or closed issues around this request.
- [ ] If this is an issue, and not a request, I have verified that this is something that is reproducible

**Device with issue:** <Pixel 2 XL running Android P beta 4, had the same issue on Oreo>
**Primary device:** <Pixel 2 XL^>
**App version:** <4.0.0.2174>
**Steps to reproduce:**

1. Go into a conversation that occurred recently, with many back and forth texts
2. Simply tap on a message like you would do to see when it was sent/received
3. Notice the message says "now" instead of ie. 11:43 am

`... more steps`

**Additional information:**

Do you want to provide any other information? Device configuration, unique cases, screenshots, anything that you think may be relevant.

![screenshot_20180729-115200](https://user-images.githubusercontent.com/17242645/43368626-d4ca3934-9325-11e8-9c4e-4bfbd10eaf7d.png)

**Comment by klinker24** [comment]
So when you long click and view the details, it shows the correct time, then? What about on the web app, it shows the correct time there? Does this "now" persist if you close and re-open the conversations? I will need more detail, sorry. I cannot verify this through the steps you have provided.

**Comment by perrymike15** [comment]
My apologies for the lacking info. When you long click, it shows the correct time with all the other information about that specific text, but even when you exit that window, it still comes back to show now. If you tap on the message to hide the timestamp and tap again to reopen it, it still says just "now". It happens on multiple conversations and after closing that conversation, even the whole app, it still does this. I don't know about on the web app, I do not use it. In the app, it either shows the correct time, or "now" even when that doesn't apply.  After a long press, the time in that window is always correct.

**Comment by klinker24** [comment]
Is it always received messages? Is it always the same situation, where your message shows the time, then the previous one does not? Is it stuck at "now" forever, or how long before they change to the absolute/correct time? I can't imagine that they are stuck like that for weeks.

**Comment by perrymike15** [comment]
Nope, it's sent and received. It's random which messages show "now", although it's mainly recent messages (past 10 mins or 10 messages, whichever occors first. More often than not, a message will show now, as apposed to the absolute time. It isn't stuck at now forever, no. After another 10 ish messages added to the conversation it'll show the absolute time, or if no other messages are added then about an hour it'll actually show correct.

**Comment by perrymike15** [comment]
The only messages this is a problem with is recent ones.

**Comment by klinker24** [comment]
Thanks for the follow-up. I don't see any reason for this and do not know why that would be, sorry. If you come up with other, or more repeatable, steps for me to reproduce this issue, feel free to comment back. I have never seen this myself.

I will continue to explore the issue.

**Comment by klinker24** [comment]
I suppose that the latest version of the app (4.0.1) didn't change anything around this? I made a few small changes.

**Comment by perrymike15** [comment]
I haven't noticed the issue since the update, or at least its less
prevalent.
